crates/cli/src/cli_args/add.rs

```diff
@@ -0,0 +1,180 @@
+use pacquet_package_manager::Add;
+use std::path::PathBuf;
+
+#[derive(Debug, Args)]
+pub struct AddDependencyOptions {
+    /// Install the specified packages as regular dependencies.
+    #[clap(short = 'P', long)]
+    save_prod: bool,
+    /// Install the specified packages as devDependencies.
+    #[clap(short = 'D', long)]
+    save_dev: bool,
+    /// Install the specified packages as optionalDependencies.
+    #[clap(short = 'O', long)]
+    save_optional: bool,
+    /// Using --save-peer will add one or more packages to peerDependencies and install them as dev dependencies
+    #[clap(long)]
+    save_peer: bool,
+}
+
+impl AddDependencyOptions {
+    /// Convert the `--save-*` flags to an iterator of [`DependencyGroup`]
+    /// which selects which target group to save to.
+    fn dependency_groups(&self) -> impl Iterator<Item = DependencyGroup> {
+        let &AddDependencyOptions { save_prod, save_dev, save_optional, save_peer } = self;
+        let has_prod = save_prod || (!save_dev && !save_optional && !save_peer); // no --save-* flags implies --save-prod
+        let has_dev = save_dev || (!save_prod && !save_optional && save_peer); // --save-peer with nothing else implies --save-dev
+        let has_optional = save_optional;
+        let has_peer = save_peer;
+        std::iter::empty()
+            .chain(has_prod.then_some(DependencyGroup::Default))
+            .chain(has_dev.then_some(DependencyGroup::Dev))
+            .chain(has_optional.then_some(DependencyGroup::Optional))
+            .chain(has_peer.then_some(DependencyGroup::Peer))
+    }
+}
+
+#[derive(Debug, Args)]
+pub struct AddArgs {
+    /// Name of the package
+    pub package: String,
+    /// --save-prod, --save-dev, --save-optional, --save-peer
+    #[clap(flatten)]
+    pub dependency_options: AddDependencyOptions,
+    /// Saved dependencies will be configured with an exact version rather than using
+    /// pacquet's default semver range operator.
+    #[clap(short = 'E', long = "save-exact")]
+    pub save_exact: bool,
+    /// The directory with links to the store (default is node_modules/.pacquet).
+    /// All direct and indirect dependencies of the project are linked into this directory
+    #[clap(long = "virtual-store-dir", default_value = "node_modules/.pacquet")]
+    pub virtual_store_dir: Option<PathBuf>, // TODO: make use of this
+}
+
+impl AddArgs {
+    /// Execute the subcommand.
+    pub async fn run(self, mut state: State) -> miette::Result<()> {
+        // TODO: if a package already exists in another dependency group, don't remove the existing entry.
+
+        let State { tarball_cache, http_client, config, package_json, lockfile } = &mut state;
+
+        Add {
+            tarball_cache,
+            http_client,
+            config,
+            package_json,
+            lockfile: lockfile.as_ref(),
+            list_dependency_groups: || self.dependency_options.dependency_groups(),
+            package: &self.package,
```

Comment:
it is not the actual package, just the package specifier. I would recommend to rename it to `package_specifier`. Also, it should be an array. It is OK if that will be done later.